src/internal_events/socket.rs

```diff
@@ -63,6 +63,27 @@ impl InternalEvent for SocketEventsReceived {
     }
 }
 
+#[derive(Debug)]
+pub struct SocketBytesSent {
+    pub mode: SocketMode,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for SocketBytesSent {
+    fn emit(self) {
+        let protocol = self.mode.as_str().to_string();
```

Comment:
Given the finite size of socket modes, couldn't we make these static strings, to avoid the allocation on each emit call?

Reply:
I see `as_str()` already emits a static str. I'm confused, but assume we have to allocate to pass the value to the `trace` macro? Is that a limitation on the macro itself, or is there an inherent need for an owned value to be passed in?

Reply:
You are absolutely correct. I copied that code from another call site to `as_str` without thinking, which also added the unnecessary `.to_string`. I'll follow up.